specification/peering/resource-manager/Microsoft.Peering/preview/2019-03-01-preview/peering.json

```diff
@@ -0,0 +1,1449 @@
+            "name": "sessionStateV6",
+            "modelAsString": true
+          }
+        },
+        "maxPrefixesAdvertisedV4": {
+          "format": "int32",
+          "description": "The maximum number of prefixes advertised over the IPv4 session.",
+          "type": "integer"
+        },
+        "maxPrefixesAdvertisedV6": {
+          "format": "int32",
+          "description": "The maximum number of prefixes advertised over the IPv6 session.",
+          "type": "integer"
+        },
+        "md5AuthenticationKey": {
+          "description": "The MD5 authentication key of the session.",
+          "type": "string"
+        }
+      }
+    },
+    "OperationListResult": {
+      "description": "The paginated list of peering API operations.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of peering API operations.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          }
+        },
+        "nextLink": {
+          "description": "The link to fetch the next page of peering API operations.",
+          "type": "string"
+        }
+      }
+    },
+    "Operation": {
+      "description": "The peering API operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "display": {
+          "$ref": "#/definitions/OperationDisplayInfo",
+          "description": "The information related to the operation.",
+          "readOnly": true
+        },
+        "isDataAction": {
+          "description": "The flag that indicates whether the operation applies to data plane.",
+          "type": "boolean",
+          "readOnly": true
+        }
+      }
+    },
+    "OperationDisplayInfo": {
+      "description": "The information related to the operation.",
+      "type": "object",
+      "properties": {
+        "provider": {
+          "description": "The name of the resource provider.",
+          "type": "string",
+          "readOnly": true
+        },
+        "resource": {
+          "description": "The type of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "operation": {
+          "description": "The name of the operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "description": {
+          "description": "The description of the operation.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "PeerAsn": {
+      "description": "The essential information related to the pee's ASN.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PeerAsnProperties",
+          "description": "The properties that define a peer's ASN.",
+          "x-ms-client-flatten": true
+        },
+        "name": {
```

Comment:
would you want to abstract the "resource" properties into a "Resource" definition and use it with "allOf" for ["Peering"](https://github.com/Azure/azure-rest-api-specs/pull/5343/files#diff-9dd14f9f472a0f883223a1a597b3be79R669) and this one? - see https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json

Reply:
give me a second to talk it out with our team.

Reply:
we made these changes along with the PeeringLocation